### ISSUE TITLE
Easy way to restrict task and all dependencies to certain workers with dask.compute()

### ISSUE THREAD

**Issue by cbyrohl** [dask/distributed#3592]
Hi there,

currently dask.compute(task, workers="worker1") will force "task" to run on "worker1" but not its dependencies. Is there an easy way to enforce this for the dependencies as well?

I guess there should be a hacky solution using task.dask.dependencies or task.dask.layers? Those return the names of the dependencies. If one can get the corresponding objects from those strings you could feed those into 'workers=(dependencies: "worker1")'.

Thanks,
Chris

**Comment by mrocklin** [comment]
No, there is not currently a way to do this.  I agree with you though that it would be nice if there was.

**Comment by jbouzaglou** [comment]
We are interested in this as well, it would really help in terms of debugging infrastructure related issues

**Comment by mrocklin** [comment]
If anyone wants to resolve this issue, that would be welcome
